starter-apps/vue/vue-template/src/components/Header.vue

```diff
@@ -0,0 +1,92 @@
+<script setup>
+import { useI18n } from 'vue-i18n'
+import { useRouter } from 'vue-router'
+import { computed } from 'vue'
+
+import AppLink from '@/components/AppLink.vue'
+import { getLocalizedPath } from '@/router/index.js'
+import { getOtherLocale } from '@/i18n/index.js'
+import HeaderBreadcrumbs from '@/components/HeaderBreadcrumbs.vue'
+import { navigateTo } from '@/utils/nav.js'
+import { HOME } from '@/config/constants.js'
+
+const router = useRouter()
+
+const { t } = useI18n()
+
+// Get the path for the other language. Used in the language toggle button.
+const getOtherLangPath = computed(() => {
+  let currentPath = router.currentRoute.value.name
+  currentPath === undefined && (currentPath = HOME)
+  if (currentPath === undefined) {
+    return
+  }
+  return getLocalizedPath(currentPath, getOtherLocale())
+})
+
+const isCurrentRoute = (routeName) => {
+  return router.currentRoute.value.name === routeName
+}
+
+// TODO: Set the top navigation elements
+const navElements = [
+  {
+    name: HOME,
+    label: t(HOME)
+  },
+  {
+    name: 'about',
+    label: t('about')
+  },
+  {
+    name: 'reportABug',
+    label: t('reportABug')
+  }
+]
+</script>
+
+<template>
+  <gcds-header skip-to-href="#main">
+    <gcds-search slot="search"></gcds-search>
+    <div slot="toggle">
+      <AppLink :to="getOtherLangPath">{{ t('langToggle') }}</AppLink>
+    </div>
```

Comment:
I believe the `toggle` slot can be replaced by using the `lang-href` prop directly on the header. To my knowledge, the slot is only available for users who prefer not to use our built-in language toggle.

Reply:
This uses the `AppLink` component for the toggle, if I remove the toggle slot, it will behave like a regular link and won't function like a single page application